### PR TITLE
Feature/te 426 fix reset application

### PR DESCRIPTION
**Pull request by gunther-bachmann**

### CODE REVIEW

File: rcp/org.testeditor.rcp4/src/org/testeditor/rcp4/handlers/ResetUIHandler.xtend [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+import org.testeditor.rcp4.Constants
+
+class ResetUIHandler {
+
+	static val logger = LoggerFactory.getLogger(ResetUIHandler)
+	@Inject MApplication application
+
+	@Execute
+	public def void resetUI(EModelService modelService, EPartService partService) {
+		val window = application.getChildren().get(0) as MWindow
```
Comment (kpanier) [inline rcp/org.testeditor.rcp4/src/org/testeditor/rcp4/handlers/ResetUIHandler.xtend]: Why need we 'as MWindow' the method getChildren return List<MWindow>

File: rcp/org.testeditor.rcp4/src/org/testeditor/rcp4/handlers/ResetUIHandler.xtend [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+		val perspectiveStack = modelService.find(Constants.MAIN_PERSPECTIVE_STACK_ID, application) as MPerspectiveStack
+		val originalPerspective = modelService.cloneSnippet(application, Constants.PERSPECTIVE_ID,
+			window) as MPerspective
+		if (originalPerspective === null) {
+			logger.warn('Could not reset UI since no original perspective has been saved (see SaveUIHandler.saveUI()).')
+			return
+		}
+
+		if (perspectiveStack.selectedElement !== null) {
```
Comment (kpanier) [inline rcp/org.testeditor.rcp4/src/org/testeditor/rcp4/handlers/ResetUIHandler.xtend]: Why triple not equals?

Comment (gunther-bachmann) [inline rcp/org.testeditor.rcp4/src/org/testeditor/rcp4/handlers/ResetUIHandler.xtend]: !== is the xtend variant of not identical, === is the equivalent of identical where == and != is intepreted as a call to compare (which does not make sense here)

File: rcp/org.testeditor.rcp4.uatests/src/main/java/org/testeditor/fixture/swt/SWTFixture.java [hunk 3]
```diff
@@ -428,25 +428,62 @@ public SWTBotCheckBox getCheckBox(String locator, SWTLocatorStrategy locatorStra
...
+		String newContent = removeLineFromString(contents, lineNumber);
+		textEditor.setText(newContent);
+		String condensedContentsAfter = condenseForLogging(textEditor.getText());
+		logger.debug("Editor contains '{}' after.", condensedContentsAfter);
+	}
+
+	private String removeLineFromString(String string, int lineNumber) {
+		String[] contentLines = string.split(System.lineSeparator());
+		if (lineNumber < 1 || lineNumber > contentLines.length) {
+			return string;
```
Comment (kpanier) [inline rcp/org.testeditor.rcp4.uatests/src/main/java/org/testeditor/fixture/swt/SWTFixture.java]: I think we should throw an exception to show the tester, that the line number is not valid for the editor content.

File: rcp/org.testeditor.rcp4.uatests/src/main/java/org/testeditor/fixture/swt/SWTFixture.java [hunk 4]
```diff
@@ -428,25 +428,62 @@ public SWTBotCheckBox getCheckBox(String locator, SWTLocatorStrategy locatorStra
...
+			}
+		}
+		return newContent.toString();
 	}
 
 	@FixtureMethod
 	public void saveActiveEditor() {
 		SWTBotEditor activeEditor = bot.activeEditor();
+		String textEditorContents = activeEditor.toTextEditor().getText();
+		String condensedContents = condenseForLogging(textEditorContents);
```
Comment (kpanier) [inline rcp/org.testeditor.rcp4.uatests/src/main/java/org/testeditor/fixture/swt/SWTFixture.java]: We use this variable only for debugging, so we can write call it in the log function.

**Pull request merged**